### PR TITLE
Fix re-rendering of list

### PR DESCRIPTION
**Pull request by tmoutinho**
This error was introduced while migrating the hooks folder.

### CODE REVIEW

File: src/hooks/useSubscriptions.ts [hunk 1]
```diff
@@ -41,18 +41,17 @@ export function useConfigSubscription(garden: Garden | null) {
 
     configSubscription.current = garden.onConfig(onConfigHandler)
 
-    return () => {
-      configSubscription.current = null
-    }
+    return () => configSubscription.current.unsubscribe()
```
Comment (reznord) [inline src/hooks/useSubscriptions.ts:44]: This should be:

```suggestion
    return () => {
      configSubscription.current.unsubscribe()
    }
```

because useEffect expects undefined during unsubscribing.

File: src/hooks/useSubscriptions.ts [hunk 2]
```diff
@@ -41,18 +41,17 @@ export function useConfigSubscription(garden: Garden | null) {
...
   }, [garden, onConfigHandler])
 
   return config
 }
 
 export function useProposalsSubscription(filters: FiltersType) {
   const { config, connector } = useGardenState()
-  const [proposals, setProposals] = useState<any[]>([])
-  const proposalsSubscription = useRef(null)
+  const [proposals, setProposals] = useState<any>([])
```
Comment (reznord) [inline src/hooks/useSubscriptions.ts:52]: why did we change the type of the proposals here and generalised it to `any`?

File: src/hooks/useSubscriptions.ts [hunk 3]
```diff
@@ -84,9 +83,7 @@ export function useProposalsSubscription(filters: FiltersType) {
       onProposalsHandler
     )
 
-    return () => {
-      proposalsSubscription.current = null
-    }
+    return () => proposalsSubscription.current.unsubscribe()
```
Comment (reznord) [inline src/hooks/useSubscriptions.ts:86]: This should be:

```suggestion
    return () => {
      proposalsSubscription.current.unsubscribe()
    }
```
because useEffect expects undefined during unsubscribing.

File: src/hooks/useSubscriptions.ts [hunk 4]
```diff
@@ -144,9 +141,7 @@ export function useProposalSubscription(
       onProposalHandler
     )
 
-    return () => {
-      proposalSubscription.current = null
-    }
+    return () => proposalSubscription.current.unsubscribe()
```
Comment (reznord) [inline src/hooks/useSubscriptions.ts:144]: This should be:

```suggestion
    return () => {
      proposalSubscription.current.unsubscribe()
    }
```

because useEffect expects undefined during unsubscribing.

File: src/logic/garden-logic.ts [hunk 5]
```diff
@@ -15,12 +15,8 @@ export default function useGardenLogic() {
...
 
   const actions = useActions()
-  const [
-    proposals,
-    filters,
-    proposalsFetchedCount,
-    blockHasLoaded,
-  ] = useProposals()
+  const [proposals, filters, proposalsFetchedCount, blockHasLoaded] =
+    useProposals()
```
Comment (reznord) [inline src/logic/garden-logic.ts:19]: since there's no change in here, can we revert the changes in this file?

File: src/hooks/useSubscriptions.ts [hunk 6]
```diff
@@ -110,8 +107,8 @@ export function useProposalSubscription(
   const [proposal, setProposal] = useState(null)
   const [loading, setLoading] = useState(true)
 
-  const rawProposalRef = useRef<unknown>(null)
-  const proposalSubscription = useRef<unknown>(null)
+  const rawProposalRef = useRef<any>(null)
+  const proposalSubscription = useRef<any>(null)
```
Comment (reznord) [inline src/hooks/useSubscriptions.ts:111]: why did we change the type from `unknown` to `any`? TS recommends to use `unknown` when you don't know what are the types of the attributes.

So, can we revert it to `unknown` like it was earlier?

**Review by reznord — CHANGES_REQUESTED** [review]
Left some suggestions and questions as well.

File: src/hooks/useSubscriptions.ts (hunk 2 above)
Comment (tmoutinho) [reply src/hooks/useSubscriptions.ts]: Done ✅

File: src/hooks/useSubscriptions.ts (hunk 6 above)
Comment (tmoutinho) [reply src/hooks/useSubscriptions.ts]: Done ✅

File: src/logic/garden-logic.ts (hunk 5 above)
Comment (tmoutinho) [reply src/logic/garden-logic.ts:19]: It was auto formatting thing 😅

**Review by reznord — APPROVED** [review]
LGTM! 👍🏻

**Review by 0xGabi — APPROVED** [review]
LGTM!
